pkg/apis/batch/validation/validation.go

```diff
@@ -665,6 +732,60 @@ func validateCompletions(spec, oldSpec batch.JobSpec, fldPath *field.Path, opts
 	return allErrs
 }
 
+func isConditionTrue(list []batch.JobCondition, cType batch.JobConditionType) bool {
+	if condition := findConditionByType(list, cType); condition != nil {
+		return condition.Status == api.ConditionTrue
+	}
+	return false
+}
+
+func findConditionByType(list []batch.JobCondition, cType batch.JobConditionType) *batch.JobCondition {
+	for i := range list {
+		if list[i].Type == cType {
+			return &list[i]
+		}
+	}
+	return nil
+}
+
+func validateIndexesFormat(indexesStr string, completions int) error {
+	if indexesStr == "" {
+		return nil
+	}
+	var lastIndex *int
+	for _, intervalStr := range strings.Split(indexesStr, ",") {
+		limitsStr := strings.Split(intervalStr, "-")
+		if len(limitsStr) > 2 {
+			return fmt.Errorf("too many parts separated by '-' in %q", intervalStr)
+		}
+		x, err := strconv.Atoi(limitsStr[0])
+		if err != nil {
+			return fmt.Errorf("cannot parse integer index: %q", limitsStr[0])
```

Comment:
The error is on converting a string to a integer. I don't usually think of that to be parsing?

Maybe "cannot convert integer to string for index".

Reply:
done